STM32F405RGTx/Core/Src/bms_entry.cpp

```diff
@@ -41,6 +41,10 @@ void _hard_fault_state_trans() {
     StateMachineThread::setState(SevereDangerFault);
 }
 
+void _set_fault_checking(bool val) {
```

Comment:
could u maybe add this as public method of `StateMachineThread` instead of here?